File: tests/fixtures/arg_refs.py
from graphql import GraphQLSchema


def build_context(request):
    return {"user_id": request.headers.get("x-user-id")}


# dict literal as return value — Gap 2 pattern
def build_config():
    return {"context": build_context, "debug": True}


# dict literal passed as positional call argument — Gap 1 pattern
def build_server_dict_arg():
    return GraphQLSchema({"context": build_context})


def build_server():
    return GraphQLSchema(context=build_context)


def pass_around(fn):
    return fn


def call_with_ref():
    pass_around(build_context)


def multi_arg(a, fn, b):
    pass

File: tests/fixtures/arg_refs.ts
import { ApolloServer } from 'apollo-server';

function buildContext(req: unknown): { userId: string } {
  return { userId: 'user-1' };
}

function buildConfig() {
  return { context: buildContext, debug: true };
}

// Apollo pattern: buildConfig returns the object, new ApolloServer receives it indirectly
const buildConfigArrow = (gateway: unknown) => ({ context: buildContext, gateway });

function buildServer() {
  const config = buildConfig();
  return new ApolloServer({ context: buildContext, schema: null });
}

// Indirect pattern: new ApolloServer(buildConfigArrow(gw)) — context is inside arrow fn return
function buildServerIndirect(gw: unknown) {
  return new ApolloServer(buildConfigArrow(gw));
}

function passAround(fn: Function) {
  return fn;
}

function callWithRef() {
  passAround(buildContext);
}
